### PR TITLE
NOISSUE - Update docs to include websocket-adapter

### PR DESCRIPTION
**Pull request by AryanGodara**
Signed-off-by: aryan <aryangodara03@gmail.com>

### Which issue(s) does this PR fix/relate to?
NOISSUE

### List any changes that modify/break current functionality
N/A

### Have you included tests for your changes?
NO

### Did you document any new/modified functionality?
YES

### Notes
N/A

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ docker-compose -f docker/docker-compose.yml up
...
+In most of the cases, HTTPS, WSS, MQTTS or secure CoAP are secure enough. However, sometimes you might need an even more secure connection. Mainflux supports mutual TLS authentication (_mTLS_) based on [X.509 certificates](https://tools.ietf.org/html/rfc5280). By default, the TLS protocol only proves the identity of the server to the client using the X.509 certificate and the authentication of the client to the server is left to the application layer. TLS also offers client-to-server authentication using client-side X.509 authentication. This is called two-way or mutual authentication. Mainflux currently supports mTLS over HTTP, MQTT, WS and MQTT over WS protocols. In order to run Docker composition with mTLS turned on, you can execute the following command from the project root:
```
Comment (drasko) [inline docs/authentication.md]: `HTTP, MQTT, WS and MQTT over WS protocols`, but `WS` after `HTTP`, it is more similar to it and belongs there.

File: docs/messaging.md [hunk 2]
```diff
@@ -66,8 +66,42 @@ The most of the notifications received from the Adapter are non-confirmable. By
...
+If you are not able to send custom headers in your handshake request, send them as
+query parameter `authorization` and `content-type`. Then your path should look like
+this `/channels/<channel_id>/messages?authorization=<thing_auth_key>&content-type=<content-type>`.
+
+If you are using the docker environment prepend the url with `ws`. So for example
+`/ws/channels/<channel_id>/messages?authorization=<thing_auth_key>&content-type=<content-type>`.
+
+### Basic nodejs example
+
+```javascript
```
Comment (drasko) [inline docs/messaging.md:85]: @AryanGodara I guess you tested this example, right?

Comment (AryanGodara) [reply docs/messaging.md:85]: I'm still working on this part, there are few typos inside.
I'm thinking of using a golang ws-client instead, I'll commit both, and later remove if it's extra

File: docs/messaging.md [hunk 3]
```diff
@@ -66,8 +66,121 @@ The most of the notifications received from the Adapter are non-confirmable. By
...
+```golang
+package main
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/gorilla/websocket"
```
Comment (rodneyosodo) [inline docs/messaging.md:114]: @AryanGodara Can you change the websocket library, it is currently archived by the gorilla team.

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: docs/messaging.md (hunk 3 above)
Comment (AryanGodara) [reply docs/messaging.md:114]: https://github.com/ultravioletrs/cocos/issues/67
Issue opened here.
Will wait for some time before updating the ws library.

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**